### PR TITLE
[AH1-18507] add RAW_BINARY for MAC (byPass String conversion as for B…

### PR DESCRIPTION
**Pull request by baeyenwo**
[AH1-18507] add RAW_BINARY for MAC (byPass String conversion as for BM-277 this causes side-effects: HEX 0x25 → 0x15)

### CODE REVIEW

**Review by thibaudledent — APPROVED** [review]

File: src/main/java/com/solab/iso8583/IsoType.java [hunk 1]
```diff
@@ -73,6 +73,11 @@ public enum IsoType {
 	 * Similar to {@link IsoType#ALPHA} but holds byte arrays instead of strings.
 	 */
 	BINARY(true, 0),
+    /**
```
Comment (YpsiBe) [inline src/main/java/com/solab/iso8583/IsoType.java:76]: Indentation is different from the rest

File: src/main/java/com/solab/iso8583/IsoValue.java [hunk 2]
```diff
@@ -465,8 +466,8 @@ public void write(final OutputStream outs, final boolean binary, final boolean f
                 return;
             }
         }
-        if (binary && (type == IsoType.BINARY || IsoType.VARIABLE_LENGTH_BIN_TYPES.contains(type))) {
-            int missing;
+        if (type == RAW_BINARY ||binary && (type == IsoType.BINARY || IsoType.VARIABLE_LENGTH_BIN_TYPES.contains(type))) {
```
Comment (YpsiBe) [inline src/main/java/com/solab/iso8583/IsoValue.java:469]: Space after the ||

File: src/test/resources/config.xml [hunk 3]
```diff
@@ -1,179 +1,186 @@
...
 
-<!-- These are the ISO headers to be prepended to the message types specified -->
-<header type="0200">ISO015000050</header>
-<header type="0210">ISO015000055</header>
-<header type="0400" ref="0200" />
-<header type="0410" ref="0210" />
-<header type="0800">ISO015000015</header>
-<header type="0810" ref="0800"/>
-<header type="0280" binary="true">ffffffff</header>
+    <!-- These are the ISO headers to be prepended to the message types specified -->
```
Comment (YpsiBe) [inline src/test/resources/config.xml:7]: Indentation

File: src/main/java/com/solab/iso8583/IsoType.java (hunk 1 above)
Comment (baeyenwo) [reply src/main/java/com/solab/iso8583/IsoType.java:76]: I'll reformat everything in a different mr - looks like the project in general doesn't follow default formatting.

File: src/test/resources/config.xml (hunk 3 above)
Comment (baeyenwo) [reply src/test/resources/config.xml]: Fixed

**Review by thibaudledent — DISMISSED** [review]

**Review by thibaudledent — APPROVED** [review]

**Pull request merged**